public/app/plugins/datasource/cloudwatch/components/LogsQueryField.tsx

```diff
@@ -33,59 +31,38 @@ export interface CloudWatchLogsQueryFieldProps
     Themeable2 {
   absoluteRange: AbsoluteTimeRange;
   onLabelsRefresh?: () => void;
-  ExtraFieldElement?: ReactNode;
+  ExtraFieldElement?: React.ReactNode;
   exploreId: ExploreId;
   query: CloudWatchLogsQuery;
 }
 
-const addPaddingToButton = css`
-  padding: 1px 4px;
-`;
-interface State {
-  hint:
-    | {
-        message: string;
-        fix: {
-          label: string;
-          action: () => void;
-        };
-      }
-    | undefined;
-}
-
-class CloudWatchLogsQueryField extends React.PureComponent<CloudWatchLogsQueryFieldProps, State> {
-  state: State = {
-    hint: undefined,
-  };
-
-  plugins: Array<Plugin<Editor>>;
+export const CloudWatchLogsQueryField = (props: CloudWatchLogsQueryFieldProps) => {
+  const { query, datasource, onChange, onRunQuery, ExtraFieldElement, data } = props;
 
-  constructor(props: CloudWatchLogsQueryFieldProps, context: React.Context<any>) {
-    super(props, context);
+  const showError = data && data.error && data.error.refId === query.refId;
+  const cleanText = datasource.languageProvider ? datasource.languageProvider.cleanText : undefined;
 
-    this.plugins = [
-      BracesPlugin(),
-      SlatePrism(
-        {
-          onlyIn: (node: Node) => node.object === 'block' && node.type === 'code_block',
-          getSyntax: (node: Node) => 'cloudwatch',
-        },
-        { ...(prismLanguages as LanguageMap), cloudwatch: syntax }
-      ),
-    ];
-  }
-
-  componentDidMount = () => {
-    const { query, datasource, onChange } = this.props;
-
-    if (onChange) {
+  const plugins: Array<Plugin<Editor>> = [
+    BracesPlugin(),
+    SlatePrism(
+      {
+        onlyIn: (node: Node) => node.object === 'block' && node.type === 'code_block',
+        getSyntax: (node: Node) => 'cloudwatch',
+      },
+      { ...(prismLanguages as LanguageMap), cloudwatch: syntax }
+    ),
+  ];
+
+  React.useEffect(() => {
+    if (onChange && !query.logGroupNames) {
```

Comment:
We noticed in the previous `componentDidMount`, onChange was being called even when query.logGroupNames existed. Do you think it would be an improvement to add the condition as above? 

@sunker @iwysiu would you have some idea?

Reply:
Yes. My only concern would be that this component for some reason was rendered before the query props were loaded. In that case, we'd overwrite the logGroupNames with default log groups. But this current code had the same problem in that case.

Reply:
Since you have the change on line 57, you can change line 58 to
```onChange({ ...query, logGroupNames: datasource.logsQueryRunner.defaultLogGroups });```